### PR TITLE
Feature/dev integrate exp1

### PR DESCRIPTION
**Pull request by KathyAgafonov**
Making adjustments to experiment 1

### CODE REVIEW

File: experiments/exp1/notes.md
Comment (GilLifshits) [inline experiments/exp1/notes.md:1]: Delete this file, I think it is empty

File: src/experiment_config.py [hunk 1]
```diff
@@ -78,5 +79,20 @@ class Experiment:
 
 
     def __post_init__(self):
-        self.EXPERIMENT_PATH = f"experiments/{self.EXPERIMENT_DATE_TIME}_{self.EXPERIMENT_ID}"
+        self.EXPERIMENT_PATH = f"experiments_logs/{self.EXPERIMENT_DATE_TIME}_{self.EXPERIMENT_ID}"
         self.SAVE_MODEL_DIRECTORY = f"{self.EXPERIMENT_PATH}/trained_model"
+
+        # Load API token from external JSON file
+        try:
+            with open("C:/Users/User/PycharmProjects/RL/src/logger/token.json", "r") as f:
```
Comment (GilLifshits) [inline src/experiment_config.py]: The path is different for each user, using src/logger/token.json (or something similar) did not work?

File: src/logger/neptune_logger.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                self.run[f"model/{metric_name}"].append(value, step=step)
+            else:
+                self.run[metric_name].append(value)
+
+    def log_actions_per_episode(self, actions_per_episode: list, car_name: str):
+        actions_str = [str(action) for action in actions_per_episode]
+        actions_str_combined = ', '.join(actions_str)
+        self.run[f"{car_name}_actions_per_episode"].log(actions_str_combined)
+
+    def log_state_to_neptune(self, car_state: np.array, car_name: str):  # TODO implement this function
```
Comment (GilLifshits) [inline src/logger/neptune_logger.py]: Remove the TODO

File: src/logger/neptune_logger.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        self.run[f"{car_name}_state"].log(car_state)
+
+    def log_model(self, model_path, model_name="trained_model"):
```
Comment (GilLifshits) [inline src/logger/neptune_logger.py]: The function is not log_model, it is upload_model

File: src/logger/neptune_logger.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+    def log_hyperparameters(self, experiment_config):
+        hyperparams = {
+            "experiment_id": experiment_config.EXPERIMENT_ID,
+            "model_type": experiment_config.MODEL_TYPE,
+            "epochs": experiment_config.EPOCHS,
+            "learning_rate": experiment_config.LEARNING_RATE,
+            "n_steps": experiment_config.N_STEPS,
+            "batch_size": experiment_config.BATCH_SIZE,
+            # "seed": experiment_config.SEED,
```
Comment (GilLifshits) [inline src/logger/neptune_logger.py]: Why in comment?

File: src/logger/neptune_logger.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+                experiment_config.CAR1_INITIAL_POSITION_OPTION_1,
+                experiment_config.CAR1_INITIAL_POSITION_OPTION_2,
+            ]),
+            "car2_initial_position_options": str([
+                experiment_config.CAR2_INITIAL_POSITION_OPTION_1,
+                experiment_config.CAR2_INITIAL_POSITION_OPTION_2,
+            ]),
+            "car1_action_fast": experiment_config.THROTTLE_FAST,
+            "car1_action_slow": experiment_config.THROTTLE_SLOW,
+            "car2_action": (experiment_config.THROTTLE_FAST + experiment_config.THROTTLE_SLOW) / 2,
```
Comment (GilLifshits) [inline src/logger/neptune_logger.py]: action and state are not hyperparams (they change each step)

File: src/experiment_config.py (hunk 1 above)
Comment (KathyAgafonov) [reply src/experiment_config.py]: The issue occurs because Python looks for files relative to the current working directory,
So if we run it from the experiments/exp1 folder, it can’t find src/logger/token.json unless that path exists from there.
But I found another method to extract the full path in a way that is not hardcoded:
`token_path = Path(__file__).resolve().parents[1] / "src" / "logger" / "token.json"`

Comment (KathyAgafonov) [reply src/experiment_config.py:87]: But I found another method to extract the full path in a way that is not hardcoded.
`token_path = Path(__file__).resolve().parents[1] / "src" / "logger" / "token.json"`

File: src/logger/neptune_logger.py (hunk 4 above)
Comment (KathyAgafonov) [reply src/logger/neptune_logger.py]: It was missing from Ariel's version and made errors, so I just put it in a comment for now.

File: src/logger/neptune_logger.py (hunk 2 above)
Comment (KathyAgafonov) [reply src/logger/neptune_logger.py]: Removed

File: src/logger/neptune_logger.py (hunk 3 above)
Comment (KathyAgafonov) [reply src/logger/neptune_logger.py]: I changed it.

File: src/logger/neptune_logger.py (hunk 5 above)
Comment (KathyAgafonov) [reply src/logger/neptune_logger.py]: Removed

File: src/logger/neptune_logger.py (hunk 4 above)
Comment (KathyAgafonov) [reply src/logger/neptune_logger.py]: I add `SEED: int = 42` to the file `experiment_config`
